app/main_dev/launch.js

```diff
@@ -359,3 +377,70 @@ export const readExesVersion = (app, grpcVersions) => {
 
   return versions;
 };
+
+// connectDaemon starts a new rpc connection to dcrd
+export const connectRpcDaemon = (mainWindow, rpcCreds) => {
+  const rpc_host = rpcCreds ? rpcCreds.rpc_host : rpchost;
+  const rpc_port = rpcCreds ? rpcCreds.rpc_port : rpcport;
+  const rpc_user = rpcCreds ? rpcCreds.rpc_user : rpcuser;
+  const rpc_pass = rpcCreds ? rpcCreds.rpc_pass : rpcpass;
+  const rpc_cert = rpcCreds ? rpcCreds.rpc_cert : rpccert;
+
+  var cert = fs.readFileSync(rpc_cert);
+  const url = `${rpc_host}:${rpc_port}`;
+  if (dcrdSocket && dcrdSocket.readyState === dcrdSocket.OPEN) {
+    return mainWindow.webContents.send("connectRpcDaemon-response", { connected: true });
+  }
+  dcrdSocket = new webSocket(`wss://${url}/ws`, {
+    headers: {
+      "Authorization": "Basic "+Buffer.from(rpc_user+":"+rpc_pass).toString("base64")
+    },
+    cert: cert,
+    ecdhCurve: "secp521r1",
+    ca: [ cert ]
+  });
+  dcrdSocket.on("open", function() {
+    console.log("CONNECTED");
+    // Send a JSON-RPC command to be notified when blocks are connected and
+    // disconnected from the chain.
+    dcrdSocket.send("{\"jsonrpc\":\"1.0\",\"id\":\"blockconnected\",\"method\":\"notifyblocks\",\"params\":[]}");
+    return mainWindow.webContents.send("connectRpcDaemon-response", { connected: true });
+  });
+  dcrdSocket.on("error", function(error) {
+    console.log("ERROR:" + error);
+    return mainWindow.webContents.send("connectRpcDaemon-response", { connected: false, error });
+  });
+  dcrdSocket.on("message", function(data) {
+    const parsedData = JSON.parse(data);
+    const id = parsedData ? parsedData.id : "";
+    switch (id) {
+    case "blockconnected":
+      // ToDo move NEWBLOCKCONNECTED ntfs for here
+      // const hex = hexToRaw(parsedData.params[0]);
+      // const newBlock = decodeConnectedBlockHeader(Buffer.from(hex));
+      break;
+    case "getinfo":
+      mainWindow.webContents.send("check-getinfo-response", parsedData.result );
+      break;
+    case "getblockchaininfo": {
+      const dataResults = parsedData.result || {};
+      const blockCount = dataResults.blocks;
+      const syncHeight = dataResults.syncheight;
+      mainWindow.webContents.send("check-daemon-response", { blockCount, syncHeight });
+      break;
+    }
+    }
+  });
+  dcrdSocket.on("close", () => {
+    console.log("DISCONNECTED");
```

Comment:
Same. Ideally this could send a signal to the GUI so that it could alert that the daemon has closed (if the gui is not performing a clean shutdown yet).

Reply:
after dcrdsocket is closed it is already on the shutdown, but I liked this idea. I will take a deeper look when refactoring startup.

Maybe that can be good for running more then one instance of decrediton

Reply:
Right. I meant that as a way of signalling error conditions (eg: dcrd crashed somehow).

Reply:
I see, I think that makes sense. We can also get when disconnecting from remotes daemons. I will take this path.